portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +27,44 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  
+  private List<String> comments;
+
+  @Override
+  public void init() {
+    comments = new ArrayList<>();
+  }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json;");
+    String json = new Gson().toJson(comments);
+    response.getWriter().println(json);
+  }
+  
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String comment = getParameter(request, "text-input", "");
+    comments.add(comment);
+
+    // Respond with the result.
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Erys Dennis!</h1>");
+    response.getWriter().println(comment);
+
+    // Redirect back to the HTML page.
+    response.sendRedirect("/index.html");
```

Comment:
optional: this would be more reusable if you provide "/index.html" as a POST parameter.  That way any page can decide where they want to be redirect after submitting